app/main.py

```diff
@@ -0,0 +1,40 @@
+import subprocess
+import time
+from contextlib import asynccontextmanager
+
+from fastapi import FastAPI
+from starlette.middleware.cors import CORSMiddleware
+from api.api_v1.api import api_router
+from core.config import settings
+from fastapi.middleware.httpsredirect import HTTPSRedirectMiddleware
+import os
+from init.db_init import init_db
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    # Sleep necessary because the db might not be up before this call, which results in a crash (restart the backend container for it to work properly though)
+
+    # auto migrate the db
+    time.sleep(3)
+    subprocess.run(["alembic", "revision", "--autogenerate"], check=True)
```

Comment:
I am not sure if autogenerating migrations inside the application is a good practise. Autogeneration may not produce correct migration. Alternatively, I suggest to run manually the migration script and commit the migration file.

Consider run the migration within the entry point script and not inside the app

Reply:
Agree on bad practice.
I can place the migration logic into a entrypoint.sh script.
I'll also push a single migration file that contains all of the models that will be pushed onto the DB in the script.
Because of this the lifespan coroutine will be removed completely.